src/java.desktop/macosx/classes/sun/lwawt/macosx/NSEvent.java

```diff
@@ -269,7 +269,12 @@ static int nsToJavaEventType(int nsEventType) {
      */
     static native char nsToJavaChar(char nsChar, int modifierFlags, boolean spaceKeyTyped);
 
-    static boolean isPopupTrigger(int jmodifiers) {
+    static boolean isPopupTrigger(int jmodifiers, int jeventType) {
+        if (jeventType != MouseEvent.MOUSE_PRESSED
+                && jeventType != MouseEvent.MOUSE_RELEASED) {
+            return false;
+        }
```

Comment:
Just to double check, are the only jeventTypes to trigger popups using mouse? Could there be other event types, maybe using keyboard for example with tab/arrows & enter?

Reply:
I believe only MouseEvents can be popup triggers. I checked in MouseEvent that there is a field for popup triggers and no such field in KeyEvent. 

The isPopupTrigger method here also previously checked button down masks already, so adding a check for the correct mouse events shouldn't be an issue even if there was a way to show a popup menu with a different event type.

Reply:
Although there's one more issue it seems where in native, after popup is shown when right mouse button (or CTRL+left mouse button) is pressed, if you release the mouse the popup vanishes which I noticed in Notes, Mail but it does not happen in JDK and popup remains open...but it seems not related to this...did you notice?

Reply:
Hmm, not sure why. I doubt it's related to this since both MOUSE_PRESSED and MOUSE_RELEASED are still popuptriggers before and after the fix. I can make a new issue for this to match native behavior.

Reply:
I've tested that interaction before and after the fix and it's the same behavior. The initial MOUSE_PRESSED will open a popup, then the subsequent MOUSE_RELEASED will do nothing. When the popup is already open, the MOUSE_PRESSED will close the popup, but the MOUSE_RELEASED will do nothing. So it's separate from this fix but it might be something to do with MOUSE_RELEASED or maybe MOUSE_DRAGGED/MOUSE_CLICKED cancelling the popup behavior. I can look to make a fix separate from this issue.